modules/gdscript/language_server/gdscript_text_document.cpp

```diff
@@ -253,6 +252,51 @@ Dictionary GDScriptTextDocument::rename(const Dictionary &p_params) {
 	return GDScriptLanguageProtocol::get_singleton()->get_workspace()->rename(params, new_name);
 }
 
+Variant GDScriptTextDocument::prepareRename(const Dictionary &p_params) {
+	lsp::TextDocumentPositionParams params;
+	params.load(p_params);
+
+	lsp::DocumentSymbol symbol;
+	lsp::Range range;
+	if (GDScriptLanguageProtocol::get_singleton()->get_workspace()->can_rename(params, symbol, range)) {
+		// Dictionary ret;
+		// ret["defaultBehavior"] = true;
+		// return ret;
```

Comment:
I assume so, but I'll ping @Booksbaum to make sure

Reply:
Looking at it, it does seem like it was used for testing the old behavior of renaming. I could be wrong though. I went ahead and removed the comments in https://github.com/godotengine/godot/pull/80973/commits/716c64184cdcf4410254547e0a5bf0ca78736901

Reply:
I deliberately left that in because that's what I want to return -- but currently the Godot VS Code extension cannot handle it.

To quote the [Commit Message](https://github.com/Booksbaum/godot/commit/d344685d75be4e4fbf3be0bba141d3302b8eaffd):
> **Change prepareRename to return range**
>
> prev: return `defaultBehavior: true`
>
> Reason for change:
> `defaultBehavior` doesn't work with current `godot-vscode-plugin`:
> Uses old `vscode-languageclient` -- and updating requires significant more changes than just updating the package.
> -> current `godot-vscode-plugin` (Version `1.3.1`) works with `range`, but not `defaultBehavior`


Though with squashing commits together it's probably best to either remove uncommented code or directly add note why it's uncommented

Reply:
> I deliberately left that in because that's what I want to return -- but currently the Godot VS Code extension cannot handle it.
> 
> To quote the [Commit Message](https://github.com/Booksbaum/godot/commit/d344685d75be4e4fbf3be0bba141d3302b8eaffd):
> 
> > **Change prepareRename to return range**
> > prev: return `defaultBehavior: true`
> > Reason for change:
> > `defaultBehavior` doesn't work with current `godot-vscode-plugin`:
> > Uses old `vscode-languageclient` -- and updating requires significant more changes than just updating the package.
> > -> current `godot-vscode-plugin` (Version `1.3.1`) works with `range`, but not `defaultBehavior`
> 
> Though with squashing commits together it's probably best to either remove uncommented code or directly add note why it's uncommented

Maybe this could be a separate PR in tandem with a PR to the godot-vscode-plugin

Reply:
The way it currently is (here in this PR) is probably even better than `defaultBehaviour` as we return the range of the symbol we later rename. While `defaultBehaviour` tells VSCode to guess what the correct symbol is -- which most likely always matches ... but might be different (-> source of truth: Godot vs. VSCode)

(And when I look at the commit: Before the check was slightly simpler -- which was probably why I used default behaviour instead (or tried to use...))